src/lib/defaultExtractor.js

```diff
@@ -12,7 +12,7 @@ export function defaultExtractor(context) {
     let results = []
 
     for (let pattern of patterns) {
-      results.push(...(content.match(pattern) ?? []))
+      results = [...results, ...(content.match(pattern) ?? [])]
```

Comment:
I don't really know the context here, but if performance is at all relevant, then you might consider doing something like this instead:

```javascript
for (let pattern of patterns) {
  for (let result of (content.match(pattern) || [])) {
    results.push(result);
  }
}
```

(didn't test it or anything, so apologies if that isn't quite right, but the point is that there's no need to iterate through `results` on every iteration of `patterns`; same principle as in this piece: https://betterprogramming.pub/the-reduce-spread-anti-pattern-fc0c1c0b23f6)

Reply:
I looked into this a bit and there's very little perf difference (in a plain program at least) when given a super large content array. There's a slight difference in heap allocations but it's tiny. The perf difference is on the order of ± 0.1ms for 1000 iterations.

Though, my testing method could be a bit flawed or V8 may have hyper optimized my test case. Either way I think it's not a big deal one way or the other.

Reply:
Huh. I'd be surprised if there's really no significant difference, since pushing only the new items each time is O(n), whereas _also_ iterating through all previously added items each time is O(n^2-ish). But yeah, could be that they've optimized it. Cheers.

Reply:
This is because array's in V8 are likely tail-allocated. This means doing something like `result = [...result, ...anything]` makes only a constant number of allocations independent of the number of items in `result`. Only the number of items in `anything` matters. If you flip the arrays around (`result = [...anything, ...result]`) it can't reuse existing memory/storage and must copy memory around. If you try this it results in times that are almost 4x slower.